### ISSUE TITLE
`sorted_imports` rule not working

### ISSUE THREAD

**Issue by pylapp** [realm/SwiftLint#5823]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

The _SwiftLint_ linter, used both in CLI and Xcode (15.3) does not trigger any warning for bad sorted imports.
I do not understand how the rule should work, and nothing is trigerred for both [our app and the Swift Package of our project](https://github.com/Orange-OpenSource/ouds-ios).

The source code bellow should, FMPOV, trigger a warning because "OUDS" alphabetically is before "SwiftUI"
```swift
import SwiftUI
import OUDS

@main
struct Showcase: App {
    ...
}
```

I tested also by inverting the imports:
```swift
import OUDS
import SwiftUI

@main
struct Showcase: App {
    ...
}
```

But no warnings.

I used _SwiftLint_ in CLI mode with the command below, but does not work

```bash
swiftlint --path Showcase/Showcase/Showcase.swift
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure): 0.52.4
* Xcode version (run `xcodebuild -version` to be sure): Xcode 15.3 / Build version 15E204a
* Installation method used (Homebrew, CocoaPods, building from source, etc): no idea ^^'
* Configuration file: [Get here here (open source project)](https://github.com/Orange-OpenSource/ouds-ios/blob/develop/.swiftlint.yml)

**Comment by SimplyDanny** [comment]
I checked your configuration file. It's actually not correctly formatted, namely the opt-in rules section `opt_in_rules` may only contain a list of rules. If these rules require additional configuration, there needs to be a dedicated section in the file that's named like the rule.

So instead of 

```yml
opt_in_rules:
  - trailing_closure:
      only_single_muted_parameter: true
```

you need to write

```yml
opt_in_rules:
  - trailing_closure

trailing_closure:
  only_single_muted_parameter: true
```

I didn't try but assume this is going to fix the issue with `sorted_imports` (and likely even other rules).

**Comment by pylapp** [comment]
Ok then I may have spammed you, I have sorry for that 🙁

I am a bit disappointed because I checked the file with online linters and also with some GenAI tool like ChatGPT to find the issue, but the thing you point was not found 😭

If you know efficient online or offline linters I am interested 😄

I will check the file in the next days, and keep you in touch.

**Comment by SimplyDanny** [comment]
> Ok then I may have spammed you, I have sorry for that 🙁
> 
> I am a bit disappointed because I checked the file with online linters and also with some GenAI tool like ChatGPT to find the issue, but the thing you point was not found 😭
> 
> If you know efficient online or offline linters I am interested 😄
> 
> I will check the file in the next days, and keep you in touch.

Your file is valid YAML, so linters won't find any issue with it. This is more about the format SwiftLint expects and that's so far only documented in the README.

**Comment by pylapp** [comment]
I applied your suggestions, and it worked like a charm!
Thank you so much!
